Makefile.shared

```diff
@@ -311,3 +311,30 @@ partialclean::
 	rm -f driver/compdynlink.mli
 	rm -f driver/compdynlink.mlopt
 
+# The native toplevel
+
+compilerlibs/ocamlopttoplevel.cmxa: $(OPTTOPLEVEL:.cmo=.cmx)
+	$(CAMLOPT) -a -o $@ $(OPTTOPLEVEL:.cmo=.cmx)
+partialclean::
+	rm -f compilerlibs/ocamlopttoplevel.cmxa
+
+# When the native toplevel executable has an extension (e.g. ".exe"),
+# provide a phony 'ocamlnat' synonym
+
+ifneq ($(EXE),)
+.PHYNY: ocamlnat
```

Comment:
Alain Frisch (2016/07/31 08:13 -0700):
> Did you try with the original rule targetting "ocamlnat" (even if it 
> produces "ocamlnat.exe" in reality)?

Do you mean something like

ocamlnat: compilerlibs/ocamlcommon.cmxa compilerlibs/ocamloptcomp.cmxa \
    compilerlibs/ocamlbytecomp.cmxa \
    compilerlibs/ocamlopttoplevel.cmxa \
    $(OPTTOPLEVELSTART:.cmo=.cmx)
	$(CAMLOPT) $(LINKFLAGS) -linkall -o $@ ocamlnat$(EXE)

?

Well to be honest, this rule that may not exactly produce its target
looks a bit odd to me, I'd even say a bit awkward.

Even if that is going to work under Cygwin which is currently a build
requirement for OCaml under Windows, I find it a pity to introduce a
solution which relies on this when it i possible to do otherwise.